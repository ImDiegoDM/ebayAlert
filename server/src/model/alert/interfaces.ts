import * as validator from 'validator'
import { ObjError } from '../../objError';

export interface SearchPhrase{
  phrase:string;
  howOften:2|10|30
}

export interface Alert{
  id?:string,
  email:string,
  phrases:SearchPhrase[]
}

export const collection = 'search_phrases';
