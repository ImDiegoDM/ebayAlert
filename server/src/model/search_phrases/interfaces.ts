export interface ISearchPhrase {
  email: string;
  phrase: string;
  howOften: 2 | 10 | 30;
}

export const collection = 'search_phrases';
