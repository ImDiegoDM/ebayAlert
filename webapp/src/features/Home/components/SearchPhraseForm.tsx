import React from 'react';
import styled from 'styled-components';
import { ISearchPhrase } from '../../../interfaces';

const Form = styled.form`
  display: flex;
  flex-direction: column;
  padding: 20px;
`;

const Label = styled.label`
  display: flex;
  flex-direction: column;
  margin-bottom: 15px;
`;

interface SearchPhraseFormProps {
  value: ISearchPhrase;
  onValueChange: (value: ISearchPhrase) => void;
  onSubmit: () => void;
}

export function SearchPhraseForm(props: SearchPhraseFormProps) {
  const value = props.value;

  return <Form onSubmit={(e) => {
    e.preventDefault();
    props.onSubmit();
  }}>
    <Label>
      Email
      <input
        value={value.email}
        required
        onChange={(e) => {
          props.onValueChange({...value, email: e.target.value});
        }}
        type="email"/>
    </Label>
    <Label>
      Phrase
      <input
        required
        value={value.phrase}
        onChange={(e) => {
          props.onValueChange({...value, phrase: e.target.value});
        }}
        type="text"/>
    </Label>
    <Label>
      How often
      <select value={value.howOften} onChange={(e) => {
        props.onValueChange({...value, howOften: (e.target.value as '2' | '10' | '30')});
      }}>
        <option value="2">2</option>
        <option value="10">10</option>
        <option value="30">30</option>
      </select>
    </Label>
    <div>
      <input type="submit"/>
    </div>
  </Form>;
}
