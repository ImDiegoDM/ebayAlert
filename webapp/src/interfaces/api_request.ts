import { AxiosError } from 'axios';

export interface IApiRequest<T= any, E= any> {
  fetching: boolean;
  fetched: boolean;
  ok: boolean;
  data?: T;
  error?: AxiosError;
}
