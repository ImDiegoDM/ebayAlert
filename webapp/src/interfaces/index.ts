export * from './api_request';
export * from './search_phrase';
