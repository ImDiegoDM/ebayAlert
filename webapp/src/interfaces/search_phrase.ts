export interface ISearchPhrase {
  phrase: string;
  email: string;
  howOften: '2' | '10' | '30';
}
