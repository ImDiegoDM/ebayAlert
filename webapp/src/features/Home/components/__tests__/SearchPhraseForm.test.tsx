import { render } from '@testing-library/react';
import React from 'react';
import { ISearchPhrase } from '../../../../interfaces';
import { SearchPhraseForm } from '../SearchPhraseForm';

const phrase: ISearchPhrase = {
  _id: 'sdfd5fa1dfad',
  email: 'test@test.com',
  howOften: '2',
  phrase: 'drone',
};

describe('<SearchBar/>', () => {
  it('should match snapshot', () => {
    const { container } = render(<SearchPhraseForm
      value={phrase}
      onValueChange={() => undefined}
      onSubmit={() => undefined}
    />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
