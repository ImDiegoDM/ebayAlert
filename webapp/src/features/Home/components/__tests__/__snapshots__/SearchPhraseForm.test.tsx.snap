// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar/> should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 20px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 15px;
}

<form
  class="c0"
>
  <label
    class="c1"
  >
    Email
    <input
      required=""
      type="email"
      value="test@test.com"
    />
  </label>
  <label
    class="c1"
  >
    Phrase
    <input
      required=""
      type="text"
      value="drone"
    />
  </label>
  <label
    class="c1"
  >
    How often
    <select>
      <option
        value="2"
      >
        2
      </option>
      <option
        value="10"
      >
        10
      </option>
      <option
        value="30"
      >
        30
      </option>
    </select>
  </label>
  <div>
    <input
      type="submit"
    />
  </div>
</form>
`;
